src/OpenSpaceToolkit/Physics/Coordinate/Frame/Providers/J2000.cpp

```diff
@@ -0,0 +1,149 @@
+
+#include <OpenSpaceToolkit/Mathematics/Geometry/3D/Transformations/Rotations/RotationMatrix.hpp>
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Utilities.hpp>
+
+#include <sofa/sofa.h>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#define DAS2R (4.848136811095359935899141e-6)
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+namespace ostk
+{
+namespace physics
+{
+namespace coord
+{
+namespace frame
+{
+namespace provider
+{
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+using ostk::physics::time::Scale ;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+                                J2000::J2000                                (   const   iau::Theory&                aTheory                                     )
+                                :   theory_(aTheory)
+{
+
+}
+
+                                J2000::~J2000                               ( )
+{
+
+}
+
+J2000*                          J2000::clone                                ( ) const
+{
+    return new J2000(*this) ;
+}
+
+bool                            J2000::isDefined                            ( ) const
+{
+    return true ;
+}
+
+iau::Theory                     J2000::getTheory                            ( ) const
+{
+    return this->theory_ ;
+}
+
+Transform                       J2000::getTransformAt                       (   const   Instant&                    anInstant                                   ) const
+{
+
+    using ostk::math::geom::d3::trf::rot::RotationMatrix ;
+
+    if (!anInstant.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Instant") ;
+    }
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("J2000") ;
+    }
+
+    double rb[3][3] ;
+
+    switch (this->theory_)
+    {
+
+        case iau::Theory::IAU_2000A:
+        {
+
+            const double EPS0 = 84381.448 * DAS2R ;
+
+            double dpsibi, depsbi, dra0, rbw[3][3] ;
+
+            iauBi00(&dpsibi, &depsbi, &dra0) ;
+
+            iauIr(rbw) ;
+            iauRz(dra0, rbw) ;
+            iauRy(dpsibi * sin(EPS0), rbw) ;
+            iauRx(-depsbi, rbw) ;
+            iauCr(rbw, rb) ;
+
+            break ;
+
+        }
+
+        case iau::Theory::IAU_2006:
+        {
+
+            double gamb, phib, psib, epsa ;
+
+            iauPfw06(DJM0, DJM00, &gamb, &phib, &psib, &epsa) ;
+            iauFw2m(gamb, phib, psib, epsa, rb) ;
+
+            break ;
+
+        }
+
+        default:
```

Comment:
Since we're not supporting the IAU_2000B case should we remove it/comment it out for now from the above files?

Reply:
I actually have an upcoming PR which will make use of IAU 2000B for TOD calculation :)

Reply:
Ah gotcha. That makes sense!